src/testdir/test_selectmode.vim

```diff
@@ -258,4 +258,45 @@ func Test_term_mouse_multiple_clicks_to_select_mode()
   bwipe!
 endfunc
 
+" Test for selectmode register overwrite
+func Test_selectmode_register()
+  " Default behavior: use unnamed register
+  new
+  call setline(1, range(1,100))
+  let save_register = getreg('"')
+  exe ":norm! v\<c-g>a"
+  call assert_equal('1', getreg('"'))
+
+  " Disable overwrite registers
+  new
+  call setline(1, range(1,100))
+  let save_register = getreg('"')
+  exe ":norm! v\<c-g>\<c-r>_a"
+  call assert_equal(save_register, getreg('"'))
+
+  " Invalid register
+  new
+  call setline(1, range(1,100))
+  let save_register = getreg('"')
+  exe ":norm! v\<c-g>\<c-r>?a"
+  call assert_equal('1', getreg('"'))
+
+  " Use unnamed register
+  new
+  call setline(1, range(1,100))
+  let save_register = getreg('"')
+  exe ":norm! v\<c-g>\<c-r>\"a"
+  call assert_equal('1', getreg('"'))
+
+  " Overwrite a register instead
+  new
+  call setline(1, range(1,100))
+  let save_register = getreg('"')
+  exe ":norm! v\<c-g>\<c-r>aa"
+  call assert_equal(save_register, getreg('"'))
+  call assert_equal('1', getreg('a'))
+
+  bw!
+endfunc
```

Comment:
* This test leaves 4 buffers, but it's enough do `new` once at first (or should do `%bw!` at the end or do `bw!` at each trial)
* Should check the line is changed to the expected just after `exe "norm! ..."`.
* Comparing `save_register` with `getreg('"')` but cannot distinguish the register "not changed" or "changed to the same value" since the first trial. Should clear the register before each trial or set different value to the line at each trial .

Reply:
Fixed.